### ISSUE TITLE
personal saltkey not saved

### ISSUE THREAD

**Issue by stephan-schwarz** [nilsteampassnet/TeamPass#1068]
Hi, 

I'm trying to evaluate this program and at first view it looks really nice.

However the biggest issue I'm having is that the personal salt key for use with the personal folders does not seem to work really well.

There are 2 scenario's:
1)
If I enable: Use the login password as SALTkey; then I can save and view the passwords.... untill I change my user password from the user profile. After I log in with the new credentials, I'm no longer able to view the passwords of the items stored in the personal folder.

2)
If I disable the above setting, and require the user to type in their saltkey manually from the menu. It simply shows a small popup "done" however after navigating to the personal folder (or refresh the site)
The message "You should enter your personal saltkey if you need to use your personal items." still appears.
The only way to get into the personal folder, is to actually use the option that you lost your salt key. Once you've entered a new saltkey, the personal folder is visible... you can add entries and view them. However once you log out, and back in again and manually type in the saltkey you just configured, it's actually not saved again and you cannot enter the personal folder since it keeps saying "You should enter your personal saltkey if you need to use your personal items."

I have tried both the master branch and the delevopment branch (2.1.24), the latter is the one I'm currently testing.

I simply installed the latest version of wamp and followed the installation guide. After enabling the personal folder feature and create a new user I already run into this.

On a sidenote, any idea why I receive the message "The LDAP extension is not activated on this server" in the LDAP options screen? I can see the checkmark in the wamp server that the php extension "php_ldap" is enabled.

Kind regards,
Stephan Schwarz

**Comment by nilsteampassnet** [comment]
​By default, PSK is only stored during your "login session" (it is a
SESSION variable).
That's mean that once you get quit, the PSK is cleaned out.​
This strategy was implemented for security reason.

If you want to keep "stored" your PSK, it is recommended to enable "store
PSK in COOKIE" from the admin settings page.
This will permit the user to store now their PSK inside a COOKIE instead of
a SESSION variable.
This strategy is for me the best one if you want your user to be able to
keep their PSK.

Please give a try to it.

**Comment by stephan-schwarz** [comment]
Hi,
Thanks for taking the time to reply.

I understand the concept you're saying. And I indeed don't want the PSK to be stored in a cookie, so I won't even bother with that.
Preferably I would want users to type in their PSK/SALTkey when they need to access their personal folder (the default configuration using a clean install).
But like I said, for some reason if I use that option, it doesn't seem to get "saved" at all. So the personal folder remains inaccessible. 

After I log on as a user account. And go to the option "Your personal saltkey > set the saltkey". A quick message box shows up "Done!" and then the personal folders still remain inaccessible.
I've attached some screenshots to make it more clear what I'm facing.

![teampass-1](https://cloud.githubusercontent.com/assets/15363012/10825460/5c29587a-7e65-11e5-97b4-6e7321834c6a.jpg)
![teampass-2](https://cloud.githubusercontent.com/assets/15363012/10825458/5c25f6da-7e65-11e5-9956-ce70c384a856.jpg)
![teampass-3](https://cloud.githubusercontent.com/assets/15363012/10825459/5c265b0c-7e65-11e5-9cc5-2b46205b12d7.jpg)


I have this same behaviour on Firefox and Chrome. Either Windows 7/Server 2012/Windows 10.
Internet Explorer 11 appears to have issues displaying the site, it cannot display the folders at all (see screenshot below).

![teampass-ie](https://cloud.githubusercontent.com/assets/15363012/10824977/2f903754-7e63-11e5-8a81-0f27b063ab52.jpg)


P.S.: For a new user account, does the SALTkey/PSK first need to be configured using the "I've lost it" option to reset your key? Or will "Set the saltkey" suffice to save the first saltkey? (both methods still yield the same results as above though).

Kind regards,
Stephan
